dali/operators/reader/loader/tar_utils_test.cc

```diff
@@ -0,0 +1,215 @@
+  archive.Read();
+  ASSERT_TRUE(archive.Eof());
+}
+
+TEST(LibTarUtilsTestSimple, LongNameIndexing) {
+  std::string filepath(dali::filesystem::join_path(std::getenv("DALI_EXTRA_PATH"),
+                                                   "db/webdataset/sample-tar/gnu.tar"));
+  TarArchive archive(TarArchive(TarArchive(FileStream::Open(filepath, false, false))));
+  std::string name_prefix(128, '#');
+  for (int idx = 0; idx < 1000; idx++) {
+    ASSERT_EQ(archive.GetFileName(), name_prefix + to_string(idx));
+    archive.NextFile();
+  }
+}
+
+void TestArchiveEntries(TarArchive& archive, const std::vector<std::string>& prefixes, int beg,
+                        int end, bool preread) {
+  for (int idx = beg; idx < end; idx++) {
+    for (int prefix_idx = 0; prefix_idx < static_cast<int>(prefixes.size()); prefix_idx++) {
+      if (preread) {
+        archive.Read();
+      }
+      ASSERT_EQ(archive.GetFileName(), to_string(idx) + prefixes[prefix_idx]);
+      bool tmp = archive.NextFile() ^
+                 (idx == end - 1 && prefix_idx == static_cast<int>(prefixes.size()) - 1);
+      ASSERT_TRUE(tmp);
+    }
+  }
+
+  ASSERT_FALSE(archive.NextFile());
+  ASSERT_EQ(archive.GetFileName(), "");
+  ASSERT_EQ(archive.GetFileSize(), 0);
+}
+
+struct SimpleTarTestsData {
+  std::string filepath;
+  bool read_ahead;
+  bool use_mmap;
+  int beg;
+  int end;
+  std::vector<std::string> prefixes;
+};
+
+class SimpleTarTests : public ::testing::TestWithParam<SimpleTarTestsData> {
+ protected:
+  TarArchive archive;
+  SimpleTarTests()
+      : archive(FileStream::Open(GetParam().filepath, GetParam().read_ahead, GetParam().use_mmap)) {
+  }
+};
+
+TEST_P(SimpleTarTests, Index) {
+  TestArchiveEntries(archive, GetParam().prefixes, GetParam().beg, GetParam().end, false);
+}
+
+TEST_P(SimpleTarTests, PostIndex) {
+  TestArchiveEntries(archive, GetParam().prefixes, GetParam().beg, GetParam().end, true);
+}
+
+TEST_P(SimpleTarTests, Contents) {
+  TAR* handle;
+  ASSERT_EQ(tar_open(&handle, GetParam().filepath.c_str(), NULL, O_RDONLY, 0, TAR_GNU), 0);
+
+  do {
+    th_read(handle);
+    vector<uint8_t> libtar_contents;
+    uint64 count = 0;
+    uint64 filesize = th_get_size(handle);
+
+    do {
+      libtar_contents.resize(libtar_contents.size() + T_BLOCKSIZE);
+      tar_block_read(handle, libtar_contents.data() + libtar_contents.size() - T_BLOCKSIZE);
+    } while ((count += T_BLOCKSIZE) < filesize);
+    libtar_contents.resize(filesize);
+
+    ASSERT_EQ(archive.Read(), libtar_contents);
+  } while (archive.NextFile());
+  ASSERT_EQ(th_read(handle), 1);
+  ASSERT_EQ(tar_close(handle), 0);
+}
+
+auto SimpleTarTestsValues() {
+  vector<SimpleTarTestsData> values;
+
+  SimpleTarTestsData filepaths[] = {
+      {dali::filesystem::join_path(std::getenv("DALI_EXTRA_PATH"),
+                                   "db/webdataset/MNIST/devel-0.tar"),
+       false,
+       false,
+       2000,
+       3000,
+       {".cls", ".jpg"}},
```

Comment:
Nitpick - this formatting is weird.
```suggestion
      { dali::filesystem::join_path(std::getenv("DALI_EXTRA_PATH"),
                                    "db/webdataset/MNIST/devel-0.tar"),
        false,
        false,
        2000,
        3000,
        {".cls", ".jpg"}},
```
^^^ with that at least you have an even number of spaces at the beginning of each line.

Reply:
OMG. Not everyone uses it - and editing such code is unnecessarily hard.